jsx/Form.js

```diff
@@ -1056,11 +1056,17 @@ class DateElement extends Component {
       maxYear = '9999';
     }
 
+    const currentDate = new Date();
+    const currentDay = ('0' + currentDate.getDate()).slice(-2);
+    const currentMonth = ('0' + (currentDate.getMonth() + 1)).slice(-2);
+
+    console.log(maxYear);
+
     // Handle date format
     let format = this.props.dateFormat;
     let inputType = 'date';
     let minFullDate = minYear + '-01-01';
-    let maxFullDate = maxYear + '-12-31';
+    let maxFullDate = maxYear + '-' + currentMonth + '-' + currentDay;
```

Comment:
this line makes sense to me. the same change could be done for line 1073 as well

Reply:
@zaliqarosli nice catch, I'll update it.

Reply:
actually, thinking about it again, i don't think this makes sense either. let's say the maxYear is 2000 as provided by the prop. then today's month being November shouldn't stop us from being able to enter a date that has December as its month.

but then let's say the maxYear is 2019, then I understand the need to limit the month to November. I think because of these two cases, you would need to add a check for whether maxYear is before today's year or equal to or greater than today's year